File: test/c/intel_x64/rdmsr.c
#include <pal/instruction/rdmsr.h>
#include <pal/msr/ia32_feature_control.h>

void test_rdmsr_compile(void)
{
    uint32_t address = PAL_IA32_FEATURE_CONTROL_ADDRESS;
    uint64_t value = pal_execute_rdmsr(address);
    pal_print_ia32_feature_control_from_value(value);
}

File: test/rust/intel_x64/src/rdmsr.rs
use pal;

pub fn test_rdmsr_compile()
{
    let address = pal::msr::ia32_feature_control::ADDRESS;
    let value = pal::instruction::execute_rdmsr(address);
    pal::msr::ia32_feature_control::print_from_value(value);
}
